.github/workflows/npmpublish.yml

```diff
@@ -0,0 +1,37 @@
+name: Node.js Package
+
+on:
+    release:
+        types: [created]
+
+jobs:
+    build:
+        runs-on: ubuntu-latest
+        steps:
+            - uses: actions/checkout@v2
+            - uses: actions/setup-node@v1
+              with:
+                  node-version: 12
+            - name: Install dependencies
+              run: npm install --frozen-lockfile
+            - name: Run tests and generate coverage report
+              run: npm run test:coverage
+            - name: Build
+              run: npm run build
+    npm-publish:
+        needs: build
```

Comment:
If this task requires a preliminary launch of the `build`, then maybe the following steps `Install dependencies` and `Build` can be skipped?
Correct me if I'm wrong, but it seems to me that in the current configuration these steps will be performed twice.